### PR TITLE
PTX-12273 Upgrade StorageCluster test with all components enabled

### PR DESCRIPTION
**Pull request by nikolaypopov**
Upgrade StorageCluster test with all components enabled

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

File: test/integration_test/basic_test.go [hunk 1]
```diff
@@ -107,6 +107,34 @@ var testStorageClusterBasicCases = []types.TestCase{
 		},
 		TestFunc: BasicUpgradeStorageCluster,
 	},
+	{
+		TestName:        "BasicUpgradeStorageClusterWithAllComponents",
```
Comment (pure-jliao) [inline test/integration_test/basic_test.go:111]: this should cover everything in the basic upgrade test case, if this test passes, we can run this test only in our pipeline, but let's keep the default upgrade case for now

File: test/integration_test/basic_test.go [hunk 2]
```diff
@@ -107,6 +107,34 @@ var testStorageClusterBasicCases = []types.TestCase{
...
+		TestrailCaseIDs: []string{"C50241"},
+		TestSpec: func(t *testing.T) interface{} {
+			objects, err := ci_utils.ParseSpecs("storagecluster/storagecluster-with-all-components.yaml")
+			require.NoError(t, err)
+			cluster, ok := objects[0].(*corev1.StorageCluster)
+			require.True(t, ok)
+			cluster.Name = "upgrade-stc-with-all-components-test"
+			alertManagerSecret, err := ci_utils.ParseSpecs("monitoring/alertmanager-secret.yaml")
```
Comment (pure-jliao) [inline test/integration_test/basic_test.go:119]: can we parse and return the stc object only in this TestSpec function here without doing the alert manager secret creation inside? we can do this in the test function after getting the stc, check stc spec if alertmanager is enabled, then parse it then create.

File: test/integration_test/basic_test.go [hunk 3]
```diff
@@ -107,6 +107,34 @@ var testStorageClusterBasicCases = []types.TestCase{
...
+			return cluster
+		},
+		ShouldSkip: func(tc *types.TestCase) bool {
+			if len(ci_utils.PxUpgradeHopsURLList) == 0 {
+				logrus.Info("--px-upgrade-hops-url-list is empty, cannot run BasicUpgradeStorageClusterWithAllComponents test")
+				return true
+			}
+			k8sVersion, _ := k8sutil.GetVersion()
+			pxVersion := ci_utils.GetPxVersionFromSpecGenURL(ci_utils.PxUpgradeHopsURLList[0])
+			return k8sVersion.GreaterThanOrEqual(k8sutil.K8sVer1_22) && pxVersion.LessThan(pxVer2_9)
```
Comment (pure-jliao) [inline test/integration_test/basic_test.go:134]: I don't remember exactly what's the case we want to skip the test. but is this still true?

File: test/integration_test/basic_test.go [hunk 4]
```diff
@@ -352,6 +380,12 @@ func BasicUpgradeStorageCluster(tc *types.TestCase) func(*testing.T) {
 			lastHopURL = hopURL
 		}
 
+		// Delete AlertManager secret, if found
```
Comment (pure-jliao) [inline test/integration_test/basic_test.go:383]: can we do this inside of `UninstallAndValidateStorageCluster`? check if alert manager is enabled then delete the secret. as I noticed the migration test may leave stale alertmanager secret as well.

File: test/integration_test/testspec/storagecluster/storagecluster-with-all-components.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+kind: StorageCluster
+apiVersion: core.libopenstorage.org/v1
```
Comment (pure-jliao) [inline test/integration_test/testspec/storagecluster/storagecluster-with-all-components.yaml:2]: how about the pvc controller? and can we add some basic metadata here? so it's still a valid object that has all the fields required, test utils should be able to update the fields.

File: test/integration_test/basic_test.go (hunk 2 above)
Comment (nikolaypopov) [reply test/integration_test/basic_test.go]: Moved

File: test/integration_test/basic_test.go (hunk 4 above)
Comment (nikolaypopov) [reply test/integration_test/basic_test.go]: Done

File: test/integration_test/testspec/storagecluster/storagecluster-with-all-components.yaml (hunk 5 above)
Comment (nikolaypopov) [reply test/integration_test/testspec/storagecluster/storagecluster-with-all-components.yaml:2]: Added `pvc-controller` annotation to the spec

File: test/integration_test/basic_test.go (hunk 3 above)
Comment (nikolaypopov) [reply test/integration_test/basic_test.go:128]: I think PX might have some sort of reqs, I honestly don't remember either, I left it as is for now

**Review by pure-jliao — APPROVED** [review]
LGTM! please test it first.

**Pull request merged**